clippy_lints/src/dbg_macro.rs

```diff
@@ -0,0 +1,51 @@
+use rustc::lint::{EarlyContext, EarlyLintPass, LintArray, LintPass};
+use rustc::{declare_tool_lint, lint_array};
+use crate::utils::span_lint;
+use syntax::ast;
+
+/// **What it does:** Checks for usage of dbg!() macro not to have it in
+/// version control.
+///
+/// **Why is this bad?** `dbg!` macro is intended as a debugging tool.
+///
+/// **Known problems:** None.
+///
+/// **Example:**
+/// ```rust,ignore
+/// // Bad
+/// dbg!(true)
+///
+/// // Good
+/// true
+/// ```
+declare_clippy_lint! {
+    pub DBG_MACRO,
+    style,
```

Comment:
I wonder if we need a whole new category for such lints at some point. We already have things like linting `println!` calls.